### PR TITLE
Change shift count overflow to be masking.

### PR DESCRIPTION
**Pull request by sunfishcode**
Per the discussion in #350, this PR changes the design to use shift count masking.

### CODE REVIEW

**Comment by titzer** [comment]
lgtm

File: Rationale.md [hunk 1]
```diff
@@ -292,6 +292,27 @@ However, since the publication of IEEE 754-2008, MIPS has added a configuration
...
+would mean divisions would have to be doubly complicated.
+
+Similarly, WebAssembly's shift operators mask their shift counts to the number
+of bits in the shifted value. Confusingly, this means that shifting a 32-bit
+value by 32 bits is an identity operation, and that a left shift is not
+equivalent to a multiplication by a power of 2 because the overflow behavior
+is different. Nevertheless, because several popular hardware architectures
+today implement this masking behavior, and those that don't can typically
+emulate it with a single extra mask instruction, and because several popular
+source languages, including JavaScript, have come to specify this behavior
```
Comment (lukewagner) [inline Rationale.md]: Maybe throw in ", Java, and C#"

**Comment by lukewagner** [comment]
lgtm

**Comment by sunfishcode** [comment]
Merging with lgtms.

File: Rationale.md [hunk 2]
```diff
@@ -292,6 +292,27 @@ However, since the publication of IEEE 754-2008, MIPS has added a configuration
...
+
+Similarly, WebAssembly's shift operators mask their shift counts to the number
+of bits in the shifted value. Confusingly, this means that shifting a 32-bit
+value by 32 bits is an identity operation, and that a left shift is not
+equivalent to a multiplication by a power of 2 because the overflow behavior
+is different. Nevertheless, because several popular hardware architectures
+today implement this masking behavior, and those that don't can typically
+emulate it with a single extra mask instruction, and because several popular
+source languages, including JavaScript and C#, have come to specify this
+behavior too, we reluctantly adopt this behavior as well.
```
Comment (dschuff) [inline Rationale.md]: The strong rationale given here makes me enthusiastic rather than reluctant about this behavior, compared to what we had before :)

Comment (taisel) [inline Rationale.md]: I'm still wondering what the tradeoffs would be for adding both shift behaviors into the languagerather than just one. Both defined, just one will tend to be slow on a specific set of hardware.

**Pull request merged**